src/OpenTelemetry.AutoInstrumentation/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,81 @@
+// <copyright file="PluginManager.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.AutoInstrumentation.Configuration;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.AutoInstrumentation.Plugins;
+
+internal class PluginManager
+{
+    private readonly IReadOnlyList<(Type Type, object Instance)> _plugins;
+
+    public PluginManager(GeneralSettings settings)
+    {
+        var plugins = new List<(Type, object)>();
+
+        foreach (var assemblyQualifiedName in settings.Plugins)
+        {
+            var type = Type.GetType(assemblyQualifiedName, throwOnError: true);
+            var instance = Activator.CreateInstance(type);
+
+            plugins.Add((type, instance));
+        }
+
+        _plugins = plugins;
+    }
+
+    public TracerProviderBuilder ConfigureTracerProviderBuilder(TracerProviderBuilder builder)
+    {
+        return ConfigureBuilder(builder, "ConfigureTracerProvider");
+    }
+
+    public MeterProviderBuilder ConfigureMeterProviderBuilder(MeterProviderBuilder builder)
+    {
+        return ConfigureBuilder(builder, "ConfigureMeterProvider");
+    }
+
+    public T ConfigureOptions<T>(T options)
```

Comment:
I think there is no need to return the `T` as AFAIK the options are always a reference type so they can be simply modified

That is why the SDK accepts `Action<Option>` and not `Func<Option, Option>` e.g. here: https://github.com/open-telemetry/opentelemetry-dotnet/blob/91a9b7c549d8e7149bc3df27a907d5e14070a232/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpMetricExporterExtensions.cs#L57

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1542/commits/c18fa0c9b02bdaf311d23c04fdc5726e91359ee9